common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/responses/resetpassword/ResetPasswordChallengeApiResponse.kt

```diff
@@ -39,6 +38,7 @@ import java.net.HttpURLConnection
  */
 class ResetPasswordChallengeApiResponse(
     @Expose override var statusCode: Int,
+    correlationId: String?,
```

Comment:
`@Expose` annotation can be added to `correlationId` as it does not contain any PII. This comment applies to all instances of correlation id in response classes.

Reply:
`@Expose` marks a field as required for JSON serialization or deserialization. Since this field is not an API response field (but rather, the value is taken from the response header, and manually set into this class), it doesn't need this annotation.
Coincidentally MSAL uses this annotation to determine whether a field should be logged or not, which in theory it can. But IMO that's not the right approach (a different annotation should be used for that).

Reply:
Ah, that feature is not enabled by default in GSON and MSAL never sets it. So this annotation is actually _only_ used to PII logging. I'll update it 👍

Reply:
Ah, GSON doesn't ignore fields with this annotation by default, it's a setting that has to be enabled. And MSAL uses this _only_ during logging. So we can indeed mark this as `@Expose`.

An improvement to our code would be to include `excludeFieldsWithoutExposeAnnotation()` when constructing our native auth responses, so fields that are actually not present in the API response such as `statusCode` and `correlationId` are ignored.

Reply:
But - I can't add `@Expose`, because of:
![image](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/assets/1315628/d10be03b-b7c6-4d5a-b165-26eaaf903ec5)

This happens because of `IApiResponse`, which hides the field and exposes only internally a setter. This is to avoid classes overwriting this value. Ideally it would be a final value, but this is not possible due to how `NativeAuthResponseHandler` is designed, which creates objects through `ObjectMapper.deserializeJsonStringToObject`. This leaves us with no change to set the correlationId param if it was a val.